core/sdk-core/src/main/java/software/amazon/awssdk/core/BytesWrapper.java

```diff
@@ -25,18 +25,25 @@
 import java.nio.charset.Charset;
 import java.util.Arrays;
 import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.annotations.SdkPublicApi;
 import software.amazon.awssdk.utils.StringUtils;
 import software.amazon.awssdk.utils.Validate;
 
-@SdkInternalApi
-abstract class BytesWrapper {
+/**
+ * A base class for {@link SdkBytes} and {@link ResponseBytes} that enables retrieving an underlying byte array as multiple
+ * different types, like a byte buffer (via {@link #asByteBuffer()}, or a string (via {@link #asUtf8String()}.
+ */
+@SdkPublicApi
+public abstract class BytesWrapper {
     private final byte[] bytes;
 
     // Needed for serialization
+    @SdkInternalApi
     BytesWrapper() {
-        this(null);
+        this(new byte[0]);
```

Comment:
Since `new byte[0]` is immutable it can be moved to a constant.

Reply:
Fixed.